FWCore/Framework/src/Event.cc

```diff
@@ -69,6 +71,14 @@ namespace edm {
         return;
       }
       gotBranchIDsFromPrevious_.resize(previousParentage->size(),false);
+      if (gotBranchIDsFromAcquire) {
+        for (auto const& branchID : *gotBranchIDsFromAcquire) {
+          addToGotBranchIDs(branchID);
+        }
+      }
+    } else if (gotBranchIDsFromAcquire) {
+      gotBranchIDsFromAcquire_ = gotBranchIDsFromAcquire;
+      gotBranchIDsFromAcquire_->clear();
```

Comment:
We get to this point (line 80) just before the acquire method is called. The module actually holds the gotBranchIDsFromAcquire container and this gives the Event a pointer to it. The container has to be in the module because the Event object will go away after acquire is done, but the contents of the container need to be remembered until produce it called (with a different Event object). At this point, it is before anything has been done on the event and the container would still contain entries from the previous event this stream processed, so it must be cleared. I will try adding comments here to clarify the logic.